src/docfx/Errors.cs

```diff
@@ -497,6 +497,9 @@ public static Error FormatInvalid(SourceInfo? source, string value, object type,
             public static Error ArrayLengthInvalid(SourceInfo? source, string propName, string criteria)
                 => new Error(ErrorLevel.Warning, "array-length-invalid", $"Array '{propName}' length should be {criteria}.", source, propName);
 
+            public static Error ConditionalCheckInvalid(SourceInfo? source, string propName, string message)
```

Comment:
consider change the error code to include `array` to indicate this validation target is array, similar to other `array-xyz` error codes.

Reply:
OK